plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+
+      output_string = []
+      for token in tokens:
+        if token in self._DELIMITERS:
+          output_string.append(token)
+          continue
+
+        token_length = len(token)
+
+        # minimum block size of base64 encoded AES-CBC is 22 bytes
+        if token_length < 22:
+          output_string.append(token)
+          continue
+
+        if not all(character in self._BASE64_ALPHABET for character in token):
+          output_string.append(token)
+          continue
+
+        rem = token_length % 4
+        if rem == 1:
+          output_string.append(token)
+          continue
+
+        while len(token) % 4:
+          token += '='
+
+        ciphertext = base64.urlsafe_b64decode(token)
+
+        if len(ciphertext) % 16:
+          output_string.append(token)
+          continue
+
+        aes_context = pycaes.context()
+        aes_context.set_key(pycaes.crypt_modes.DECRYPT, aes_key)
+        plaintext = pycaes.crypt_cbc(
+            aes_context, pycaes.crypt_modes.DECRYPT,
+            self._AES_IV, ciphertext)
+
+        unpadder = padding.PKCS7(128).unpadder()
+        try:
+          data = unpadder.update(plaintext) + unpadder.finalize()
+        except ValueError:
+          output_string.append(token)
+          continue
+
+        try:
+          output_string.append(data.decode('utf-16-le'))
+        except UnicodeError:
+          output_string.append(data)
+
+      output_strings.append(''.join(output_string))
+    return output_strings
+
+  def _ExtractStringsFromParameters(self, buffer):
+    """Extracts strings from raw parameter bytes.
+
+    Args:
+      buffer (bytearray): the buffer.
+      aes_key (bytes): the AES decryption key from general.keystore.
+      obfuscated_string_map (dict[str, str]): the mapping of obfuscated string
+          values to their original values.
+
+    Returns:
+      list[str]: strings parsed from the raw parameter bytes.
+    """
+    pascal_string_map = self._GetDataTypeMap('pascal_string')
+
+    parse_offset = 0
+    string_array = []
+
+    while parse_offset < len(buffer):
+      try:
+        pascal_string = self._ReadStructureFromByteStream(
+            buffer[parse_offset:], 0, pascal_string_map)
+
+        if len(pascal_string.value) and all(
+            c < 128 for c in pascal_string.value):
+          decoded_string = pascal_string.value.decode('utf-8')
+          string_array.append(decoded_string)
+          parse_offset += 4 + pascal_string.length
+        else:
+          parse_offset += 4
+      except errors.ParseError:
+        parse_offset += 4
+    return string_array
+
+  def _ProcessRawParameters(
+      self, raw_parameters_data, aes_key, obfuscated_string_map):
+    """Processes (extracts and decrypt/deobfuscate) the raw parameters.
+
+    Args:
+      raw_parameters_data (bytes): the raw parameters data from a log entry.
+      aes_key (Optional[bytes]): the AES key.
```

Comment:
parameter is not defined as a kwargs, but code below treats it as optional, either change the function call or the docstring and explain why aes_key can be None.

Reply:
Updated function call and docstring